platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/LenientPlatformGraphResolveState.java

```diff
@@ -179,26 +178,6 @@ public List<? extends VariantGraphResolveState> getVariantsForAttributeMatching(
         @Nullable
         @Override
         public VariantGraphResolveState getLegacyVariant() {
-            return doGetVariantByConfigurationName(Dependency.DEFAULT_CONFIGURATION);
-        }
-
-        @Nullable
-        @Override
-        public VariantGraphResolveState getVariantByConfigurationName(String name) {
-            DeprecationLogger.deprecateBehaviour("Selecting a variant by configuration name from a non-Ivy external component.")
-                .willBecomeAnErrorInGradle9()
-                .withUpgradeGuideSection(8, "selecting_variant_by_configuration_name")
-                .nagUser();
-
-            return doGetVariantByConfigurationName(name);
-        }
-
-        @Nullable
-        private VariantGraphResolveState doGetVariantByConfigurationName(String name) {
-            if (!name.equals(Dependency.DEFAULT_CONFIGURATION)) {
-                return null;
-            }
-
```

Comment:
❓ Does this still leave `getLegacyVariant` as `@Nullable` or should this be changed?